templates/terraform/processing-project/bigquery.tf

```
@@ -0,0 +1,17 @@
+resource google_bigquery_dataset dataset {
+  provider = google.target
+  dataset_id = "bigquery-dataset-dev"
```

Naming nitpicks: `dev` (vs. `prod`) is a project-level setting, I wouldn't include it here. `bigquery` and `dataset` are also redundant.

I'd suggest an alternative:
```suggestion
  dataset_id = "monster_staging_data"
```

---

I totally agree, and just updated based on these comments 👍